src/fastify-plugins/maps/index.js

```diff
@@ -0,0 +1,64 @@
+import fp from 'fastify-plugin'
+
+import { PLUGIN_NAME as MAPEO_STATIC_MAPS } from './static-maps.js'
+import { NotFoundError, getFastifyServerAddress } from '../utils.js'
+
+export const PLUGIN_NAME = 'mapeo-maps'
+
+export const plugin = fp(mapsPlugin, {
+  fastify: '4.x',
+  name: PLUGIN_NAME,
+  decorators: { fastify: ['mapeoStaticMaps'] },
+  dependencies: [MAPEO_STATIC_MAPS],
+})
+
+/**
+ * @typedef {object} MapsPluginOpts
+ * @property {string} [prefix]
+ */
+
+/** @type {import('fastify').FastifyPluginAsync<MapsPluginOpts>} */
+async function mapsPlugin(fastify, opts) {
```

Comment:
Nit: does this need to be async? (Same comment applies for `routes`.)

Reply:
doesn't need to be but then you'd have to write the plugin using a callback approach, which i think more or less pans out the same way. we've been using the `async` approach for uniformity if anything i guess